server/src/test/java/org/apache/kafka/server/ClientMetricsManagerTest.java

```diff
@@ -919,4 +922,101 @@ public void testPushTelemetryConcurrentRequestAfterSubscriptionUpdate() throws U
         // 1 request should fail with throttling error.
         assertEquals(1, throttlingErrorCount);
     }
+
+    @Test
+    public void testCacheEviction() throws UnknownHostException, InterruptedException {
+        Properties properties = new Properties();
+        properties.put("metrics", ClientMetricsConfigs.ALL_SUBSCRIBED_METRICS_CONFIG);
+        properties.put(ClientMetricsConfigs.PUSH_INTERVAL_MS, "100");
+        clientMetricsManager.updateSubscription("sub-1", properties);
+
+        GetTelemetrySubscriptionsRequest request = new GetTelemetrySubscriptionsRequest.Builder(
+            new GetTelemetrySubscriptionsRequestData(), true).build();
+
+        GetTelemetrySubscriptionsResponse response = clientMetricsManager.processGetTelemetrySubscriptionRequest(
+            request, ClientMetricsTestUtils.requestContext());
+        assertEquals(Errors.NONE, response.error());
+
+        assertNotNull(clientMetricsManager.clientInstance(response.data().clientInstanceId()));
+        assertEquals(1, clientMetricsManager.expirationTimer().size());
+        // Advance the clock to trigger cache eviction, cache expiry should be 100 * 3 = 300 ms.
+        clientMetricsManager.expirationTimer().advanceClock(300);
+        assertTimeoutPreemptively(Duration.ofMillis(300), () -> {
```

Comment:
300ms does seem rather tight for a reliable test in general. Just checking that you are confident that this isn't a flaky test in the making.

Reply:
I ran the test with @RepeatedTest for 500 times and it didn't fail for single time. The reason is that advance clock is blocking and then additional 300ms is there to complete 200 ms of tick time in hashed wheel timer. So there is a buffer of 100ms.

Reply:
I looked at it again and seems we can move away with advanceClock as background thread will enforce that. But we need to have a buffer which I have kept as 50ms so the test do not add any flakiness. Made a change, please check now.

Reply:
Though I cannot see the tests fail when waiting for 350ms but I did see a failure in PR run hence kept the preemptive wait time of 600ms which will not make this test flaky. Though that's the maximum wait time.